core/api/src/main/java/org/trellisldp/api/Resource.java

```diff
@@ -133,6 +143,18 @@ public String toString() {
      */
     Instant getModified();
 
+    /**
+     * Get the revision or state tag of the resource.
+     *
+     * @apiNote the return value of this method should be unique to the state of the resource.
+     *          That is, it should be unique across all resources on the server, and whenever
+     *          a given resource changes, this revision tag should also change. To create this
+     *          value, an implementation may simply generate a hash of the identifier and
+     *          modification date.
+     * @return the revision tag of the resource
+     */
+    String getRevision();
```

Comment:
Hm. Better `Optional<String>` here? Also, the hash thing again. If the purpose of this member is solely to _differentiate_ (not, for example, to abbreviate _and_ differentiate, or to distribute), I don't see the value of hashing.

Reply:
The problem with `Optional<String>` would be that the HTTP layer would then need to handle the ETag generation. We could re-add the `EtagGenerator` interface for cases where the `Resource` doesn't respond with a revision value. Note: at present, I'm assuming that this response will always be non-null

Reply:
Ok, but that is not what you have documented here at all. This isn't `getRevision`, it's `getETag`. That is a different move entirely, and if you want to do that we better document it. Nothing in this PR gave me to understand that this method is expected to produce an RFC7232 ETag.

Reply:
I'm going to re-work this a bit so that `::getRevision` is not expected to return a conforming ETag value

Reply:
Right, see https://github.com/trellis-ldp/trellis/pull/503#discussion_r321347455 and my reply. I think you're making the right move.